### PR TITLE
Menu Widget Documentation

### PR DESCRIPTION
**Pull request by wmdev0808**
-- add documentation for menu widget

### CODE REVIEW

File: SUMMARY.md [hunk 1]
```diff
@@ -121,6 +121,7 @@
 * [Input](widget-reference/input.md)
 * [List](widget-reference/list.md)
 * [Maps](widget-reference/maps.md)
+* [Menu](widget-reference/menu.md)
```
Comment (somangshu) [inline SUMMARY.md:124]: menu-button

File: widget-reference/menu.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+---
+description: >-
+  Menus are used to represent a set of actions in a group.
```
Comment (somangshu) [inline widget-reference/menu.md:3]: Menu buttons*

File: widget-reference/menu.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+## Properties
+| Property      | Description           |
+| :------------ | :-------------------- |
+| **label** | Sets the label of a menu. |
+| **backgroundColor** | Sets the background color of the widget. |
+| **textColor** | Sets the text color of the widget. |
+| **isDisabled** | Disables input to the widget. |
+| **isVisible** | Controls the visibility of the widget. |
+| **isCompact** | Decides if the widget is in compact mode. |
+| **menuItems** | Array of menu items in the dropdown list. Every menu item has the same properties as the top-level properties. |
```
Comment (somangshu) [inline widget-reference/menu.md:19]: We need to add documentation for the menuItem settings as well.

**Review by somangshu — CHANGES_REQUESTED** [review]

**Review by somangshu — APPROVED** [review]
LGTM.

@vihar please check

**Comment by vihar** [comment]
We'll need to add GIF's here, can I have the link to deploy-preview?

**Comment by somangshu** [comment]
@vihar [here](https://github.com/appsmithorg/appsmith/pull/5334) is the menu button widget PR. Let me know if you need anything else

**Pull request merged**